### PR TITLE
bgp,docs: deprecate MetalLB documentation, promote BGP-CP

### PR DESCRIPTION
**Pull request by ldelossa**
This commit moves the MetalLB BGP solution to depcreated and promotes
the BGP-CP feature in our docs.

Signed-off-by: Louis DeLosSantos <louis.delos@isovalent.com>

```release-note
Update BGP related documentation to reflect feature status. 
```

### CODE REVIEW

**Review by aanm — APPROVED** [review]

**Review by rastislavs — APPROVED** [review]

**Comment by ldelossa** [comment]
Going to assume doc only change did not break Travis CI here. 

No e2e tests necessary and Docs linter is passing.

File: Documentation/network/bgp.rst [hunk 1]
```diff
@@ -6,9 +6,9 @@
 
 .. _bgp:
 
-**********
-BGP (beta)
-**********
+****************
+BGP (deprecated)
```
Comment (joestringer) [inline Documentation/network/bgp.rst:10]: nit, I'd probably consider making this a bit clearer by declaring it as MetalLB-based BGP Control Plane just to make it clearer how it differs from the new one. Feel free to follow up on that afterwards.

**Review by joestringer — APPROVED** [review]

File: Documentation/network/bgp.rst (hunk 1 above)
Comment (ldelossa) [reply Documentation/network/bgp.rst:10]: follow up PR sent, check ya review queue :)

**Pull request merged**